frontend/pages/admin/TeamManagementPage/TeamDetailsWrapper/TeamSettings/TeamSettings.tsx

```diff
@@ -150,30 +163,50 @@ const TeamSettings = ({ location, router }: ITeamSubnavProps) => {
       ...DEFAULT_USE_QUERY_OPTIONS,
       enabled: isRouteOk && !!teamIdForApi,
       select: (data) => data.team,
-      onSuccess: (teamConfig) => {
+      onSuccess: (tC) => {
         setFormData({
           // host expiry settings
           teamHostExpiryEnabled:
-            teamConfig?.host_expiry_settings?.host_expiry_enabled ?? false,
+            tC?.host_expiry_settings?.host_expiry_enabled ?? false,
           teamHostExpiryWindow:
-            teamConfig?.host_expiry_settings?.host_expiry_window ?? "",
+            tC?.host_expiry_settings?.host_expiry_window ?? "",
           // host status webhook settings
           teamHostStatusWebhookEnabled:
-            teamConfig?.webhook_settings?.host_status_webhook
+            tC?.webhook_settings?.host_status_webhook
               ?.enable_host_status_webhook ?? false,
           teamHostStatusWebhookDestinationUrl:
-            teamConfig?.webhook_settings?.host_status_webhook
-              ?.destination_url ?? "",
+            tC?.webhook_settings?.host_status_webhook?.destination_url ?? "",
           teamHostStatusWebhookHostPercentage:
-            teamConfig?.webhook_settings?.host_status_webhook
-              ?.host_percentage ?? 1,
+            tC?.webhook_settings?.host_status_webhook?.host_percentage ?? 1,
           teamHostStatusWebhookWindow:
-            teamConfig?.webhook_settings?.host_status_webhook?.days_count ?? 1,
+            tC?.webhook_settings?.host_status_webhook?.days_count ?? 1,
         });
       },
     }
   );
 
+  useEffect(() => {
+    if (isInitialTeamConfig) {
+      setPercentageHostsDropdownOptions(
+        getCustomDropdownOptions(
+          HOST_STATUS_WEBHOOK_HOST_PERCENTAGE_DROPDOWN_OPTIONS,
+          teamConfig?.webhook_settings.host_status_webhook.host_percentage ?? 1,
+          (val) => `${val}%`
+        )
+      );
+
+      setWindowDropdownOptions(
+        getCustomDropdownOptions(
+          HOST_STATUS_WEBHOOK_WINDOW_DROPDOWN_OPTIONS,
+          teamConfig?.webhook_settings.host_status_webhook.days_count ?? 1,
+          (val) => `${val} day${val !== 1 ? "s" : ""}`
+        )
+      );
+    }
+    // no need for isInitialTeamConfig dependence, since this effect should only run on initial
+    // config load
+  }, [teamConfig]);
+
```

Comment:
`useEffect` and stateful management of dropdown options are very helpful here as they are set not by props passed to the component (as in the global setting, which makes it easy to avoid effects), but set by the team config response from the teams API.